Raspi/Test/serial_test.py

```diff
@@ -0,0 +1,55 @@
+import argparse
+import os
+import paho.mqtt.client as mqtt
+
+from das_data_generator import send_csv_data, send_fake_data
+
+# Arguments
+parser = argparse.ArgumentParser(description='DAS serial test script', add_help=True)
+parser.add_argument('-t', '--time', action='store', type=int, default=1, help="length of time to send data")
+parser.add_argument('-r', '--rate', action='store', type=float, default=0.5, help="rate of data in seconds")
+parser.add_argument('--host', action='store', type=str, default="localhost", help="address of the MQTT broker")
+parser.add_argument('-f', '--file', action='store', type=str, help="The csv file to replay. If not specified, makes up data.")
+parser.add_argument('-j', '--jump', action='store', type=int, default=0, help="Starts replaying from a specified time (in seconds)")
+
+def on_connect(client, userdata, flags, rc):
+    print("Connected to MQTT broker with result code " + str(rc))
+
+def run(args):
+    # Check that we are running on a unix system.
+    # os.openpty and os.ttyname are only available on unix (NOT windows)
+    if os.name != "posix":
+        raise Exception("Serial test only works on Unix systems. If you are on Windows, perhaps try using WSL.")
+
+    # Create pseudo-terminal pair
+    master, slave = os.openpty()
```

Comment:
Maybe try `pty.openpty` instead? It has some extra fallbacks in case it fails

Reply:
@khanguslee can you try this? Should be able to replace `os.openpty()` with `pty.openpty()` and run

Reply:
Unfortunately I get the same problem :(